### PR TITLE
Fix #1594 Alexa webhook for priority label

### LINKED ISSUE #1594: Implement alexa webhook for priority labels
This is the final part of #1533. Use the result db from #1591 and put the label from #1592 to created issue.

### PR DESCRIPTION
**Pull request by MDTsai**
Addd alexa topsite DB
Find priority label for domain name or less-level domain in DB

### CODE REVIEW

**Comment by MDTsai** [comment]
r? = @karlcow

File: webcompat/webhooks/helpers.py [hunk 1]
```diff
@@ -78,3 +82,25 @@ def get_payload_signature(key, payload):
...
+        for site in site_db.query(Site).filter_by(url=host_name):
+            return 'priority-{}'.format(priorities[site.priority-1])
+
+        # No host_name in DB, find less-level domain (>2)
+        # If host_name is lv4.lv3.example.com, find lv3.example.com/example.com
+        domains = host_name.split('.')
+        for i in list(reversed(range(2, len(domains)))):
+            shorten_domain_name = '.'.join(domains[(len(domains)-i):])
+            for site in site_db.query(Site).filter_by(url=shorten_domain_name):
+                return 'priority-{}'.format(priorities[site.priority-1])
```
Comment (karlcow) [inline webcompat/webhooks/helpers.py]: I would rename `host_name` as `hostname`

```python
>>> hostname = 'www.foo.bar.example.com'
>>> subparts = hostname.split('.')
>>> subparts
['www', 'foo', 'bar', 'example', 'com']
>>> for part in subparts:
...     print part
... 
www
foo
bar
example
com
>>> for i, part in enumerate(subparts):
...    print i, part
... 
0 www
1 foo
2 bar
3 example
4 com
>>> '.'.join(subparts[1:])
'foo.bar.example.com'
>>> '.'.join(subparts[2:])
'bar.example.com'
>>> '.'.join(subparts[3:])
'example.com'
>>> ['.'.join(subparts[i:]) for i, subpart in enumerate(subparts) if 0<i<hostname.count('.')]
['foo.bar.example.com', 'bar.example.com', 'example.com']
>>> hostname.count('.')
4
>>> hostname
'www.foo.bar.example.com'
>>> domains = ['.'.join(subparts[i:]) for i, subpart in enumerate(subparts) if 0<i<hostname.count('.')]
>>> # then you can iterate on domains for doing the query.
```

Something like:

```python
        subparts = hostname.split('.')
        domains = ['.'.join(subparts[i:])
                   for i, subpart in enumerate(subparts)
                   if 0 < i < hostname.count('.')]
        for domain in domains:
            # do the query
```

Comment (MDTsai) [inline webcompat/webhooks/helpers.py]: I do the 1-pass test today, with full host name matching, only 294/7840 issues found. Compared with 2-passes test before is 3212/7348.
For me I choose to keep 2-passes, and collect black list (manually or we can use alexa site info API to know which domain we should always not do 2-passes)

File: webcompat/db/__init__.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy import Column
 from sqlalchemy import String
+from sqlalchemy import Integer
 from sqlalchemy.orm import scoped_session
 from sqlalchemy.orm import sessionmaker
 from hashlib import sha512
```
Comment (MDTsai) [inline webcompat/db/__init__.py]: I move hashlib and uuid to default part.

File: webcompat/webhooks/helpers.py [hunk 3]
```diff
@@ -78,3 +82,25 @@ def get_payload_signature(key, payload):
     '''Compute the payload signature given a key.'''
     mac = hmac.new(key, msg=payload, digestmod=hashlib.sha1)
     return mac.hexdigest()
+
+
+def extract_priority_label(body):
```
Comment (MDTsai) [inline webcompat/webhooks/helpers.py]: No blank lines in function code now. Wait topsites.py to deploy and make a new DB, then we can merge this PR.

Comment (karlcow) [inline webcompat/webhooks/helpers.py]: 👍 Thanks @MDTsai

**Comment by miketaylr** [comment]
Adding do not merge label until we have Alexa API credentials in place.

**Pull request merged**